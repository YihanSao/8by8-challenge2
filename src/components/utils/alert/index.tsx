export { Alert } from './alert';
export { useAlert } from './use-alert';
