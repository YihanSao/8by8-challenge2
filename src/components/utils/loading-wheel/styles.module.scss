@use '../../../styles/partials' as *;

.transparent_overlay {
  position: fixed;
  top: $navigation-height;
  left: 0;
  width: 100vw;
  height: calc(100vh - $navigation-height);
  background-color: rgba(58, 58, 58, 0.5);
  animation: fade-in 200ms linear;
  z-index: 100;
}

.spinner_container {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  animation: fade-in 200ms linear;
  z-index: 100;
}

.spinner {
  animation: spin 1s linear infinite;
}

@keyframes fade-in {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes spin {
  from {
    rotate: 0;
  }
  to {
    rotate: 360deg;
  }
}
