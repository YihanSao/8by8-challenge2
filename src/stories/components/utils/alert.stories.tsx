import { Meta, StoryObj } from '@storybook/react';
import { Alert, useAlert } from '@/components/utils/alert';
import { GlobalStylesProvider } from '@/stories/global-styles-provider';

const meta: Meta<typeof Alert> = {
  component: Alert,
};

export default meta;

type Story = StoryObj<typeof Alert>;

function ShowAlerts() {
  const { alertRef, showAlert } = useAlert();

  const showErrorAlert = () => {
    showAlert('Your error message here.', 'error');
  };

  const showSuccessAlert = () => {
    showAlert('Your success message here.', 'success');
  };

  return (
    <GlobalStylesProvider>
      <Alert ref={alertRef} />
      <button onClick={showErrorAlert}>Show Error Alert</button>
      <button onClick={showSuccessAlert}>Show Success Alert</button>
    </GlobalStylesProvider>
  );
}

export const Default: Story = {
  render: ShowAlerts,
};
