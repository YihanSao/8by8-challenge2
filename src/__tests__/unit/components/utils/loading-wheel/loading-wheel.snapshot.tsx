import '@testing-library/jest-dom';
import { render, cleanup } from '@testing-library/react';
import { LoadingWheel } from '@/components/utils/loading-wheel';

describe('LoadingWheel', () => {
  afterEach(cleanup);

  it('renders the Spinner unchanged.', () => {
    const { container } = render(<LoadingWheel />);
    expect(container).toMatchSnapshot();
  });
});
