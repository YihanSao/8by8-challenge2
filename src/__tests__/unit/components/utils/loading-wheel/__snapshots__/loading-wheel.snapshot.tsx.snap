// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingWheel renders the Spinner unchanged. 1`] = `
<div>
  <div
    class="transparent_overlay"
  />
  <div
    class="spinner_container"
  >
    <img
      alt="loading"
      class="spinner"
      data-nimg="1"
      decoding="async"
      height="82"
      loading="lazy"
      src="/_next/image?url=%2Fstatic%2Fimages%2Fcomponents%2Fspinner%2Fspinner.png&w=256&q=75"
      srcset="/_next/image?url=%2Fstatic%2Fimages%2Fcomponents%2Fspinner%2Fspinner.png&w=96&q=75 1x, /_next/image?url=%2Fstatic%2Fimages%2Fcomponents%2Fspinner%2Fspinner.png&w=256&q=75 2x"
      style="color: transparent;"
      width="82"
    />
  </div>
</div>
`;
